### PR TITLE
Add a developer console in the menu

### PR DESCRIPTION
**Pull request by acmcarther**
Its got little fanfare, but it supports full keyboard (i think), along with arrow keys, backspace and delete.

Theres also a command log included for #72, but its unused.

Implements #66

Needs refactoring

@rschifflin

### CODE REVIEW

**Comment by acmcarther** [comment]
test_console

**Comment by acmcarther** [comment]
Much of this code is quite error prone, and deserves tests. I'll not submit it without at least _some_ tests.

File: prototype2/client/src/engine/control/console.rs [hunk 1]
```diff
@@ -0,0 +1,272 @@
...
+  Character(char),
+  Motion(CharMotion),
+  Action(CharAction),
+  Invalid,
+}
+
+fn input_to_char_event(code: VirtualKeyCode, shift: bool) -> CharEvent {
+  use glutin::VirtualKeyCode::*;
+
+  match (shift, code) {
```
Comment (rschifflin) [inline prototype2/client/src/engine/control/console.rs]: something about these mechanical mappings make me feel like they should live in their own file, some statically defined dictionary. just a thought

Comment (acmcarther) [inline prototype2/client/src/engine/control/console.rs]: Done, though its kinda ugly. Its more future proof though.

**Comment by acmcarther** [comment]
I actually can't write tests until I extract the behavior from the services. Its literally impossible to instantiate a RunArg. 

welp

**Pull request merged**